### ISSUE TITLE
Data collection issues

### ISSUE THREAD

**Issue by kdygfxb** [physical-superintelligence-lab/Psi0#44]
Hello, I would like to ask some questions about fine-tuning data.

What should the data collection frequency be, and what post-processing is required for the collected results?
Currently, I'm doing the task of picking up small cubes and putting them into a plate.When I replay the state data, the robot can perform actions normally. However, when I replay the action data, the robot's actions are quite abnormal. Will this have a significant impact on training?
The model I trained using the finetune-real-psi0.sh script performed significantly differently in actual deployment than I expected.

Looking forward to your reply

**Comment by kdygfxb** [comment]
replay;

https://github.com/user-attachments/assets/1be4727b-e6c4-4330-9ff9-9c7f71e463ce

**Comment by Jinghy123** [comment]
Hi @kdygfxb , thanks for your questions!

First, I wanted to clarify — is the video you shared showing action replay? From my side, it doesn’t look like there’s a major issue. There is a slight sense of stuttering: did you also observe this during teleoperation, or does it only appear when replaying actions? Overall, this doesn’t seem like a serious problem.

For teleoperation, it’s important that the control loop runs at at least 50 Hz to maintain stability (especially for the lower body). For data collection, around 30 Hz is sufficient, and you don’t need additional post-processing.

During replay, the behavior from action and state should not differ too much, so if the gap is large, it might be worth double-checking synchronization or control settings.

Regarding the model deployment issue, please refer to issue #39. Also, make sure that during data collection the RGB view clearly captures the task, and that during deployment the robot setup (position, viewpoint, etc.) is similar to the data collection setup, to avoid completely out-of-distribution scenarios.

Let me know your observations!

**Comment by songlin** [comment]
Hi @kdygfxb,

Thanks for reporting the issue. 

Can you further clarify how the the model performed differently than you expected?  Do you mean the model does not work? For this pick cube task, psi0 should perform very well based on our experience.

Best,
Songlin

**Comment by kdygfxb** [comment]
感谢您的回复， 这是我对于state数据的回放，而对于action数据， 机器人完全无法复现夹起小方块的动作，我是使用pico进行遥操采集， 对于state和action数据的明显差异， 我应该如何处理action数据呢， 我是否能够使用state平移t+1直接作为action呢

**Comment by songlin** [comment]
Yes, you can use the future state as actions to train the Psi0 model, and it should work. However, your pico teleoperation pipeline may need refinement. Because, in general, the teleoperated data should be able to replay successfully with high probability.

**Comment by kdygfxb** [comment]
I uploaded a demonstration of a model's performance on a real device on [](https://youtu.be/B2wwD9_1yjs) .  The model uses state data shifts as action data for training, with single-GPU training, a batch size of 128, and 20,000 epochs. All other parameters remained unchanged. However, the model's performance doesn't seem to meet expectations. Should I make any modifications to the training parameters?

**Comment by kdygfxb** [comment]
the link is https://youtu.be/AtFiDUxSLxg 

Looking forward to your reply

**Comment by kdygfxb** [comment]
I have another question. My video capture resolution is 1280*720, but I noticed that the sample data in the RealData repository is 640*480. I'd like to ask if this has a significant impact, and whether I need to recapture the data.

**Comment by songlin** [comment]
Hi @kdygfxb,

Thanks for sharing the video.

The policy indeed behaves abnormally.  I think your training parameters are fine, and the image resolution should not be a problem, as we did experiments on the Dexmate Vega robot with a 1280x720 image, and it worked well.

Based on the current situation, I strongly suggest you go through the checklist I mentioned in another issue https://github.com/physical-superintelligence-lab/Psi0/issues/39#issuecomment-4330572131.

Specifically, 

1. Check the replay is fine (but you can skip this step for now, as you can replay with the next state)
2. Monitor training loss. Can you provide the training curve?
3. Open-loop Evaluations. Please refer to [examples/simple/openloop_eval.ipynb](https://github.com/physical-superintelligence-lab/Psi0/blob/main/examples/simple/openloop_eval.ipynb)
4. Then deploy the model, and send the ground truth images/states to the server to see if the robot moves smoothly.

You might skip the steps that you feel confident about.

Best,
Songlin

**Comment by kdygfxb** [comment]
Thank you for your reply. I will conduct an open-loop evaluation of my model and upload the results later.

**Comment by kdygfxb** [comment]
<img width="961" height="772" alt="Image" src="https://github.com/user-attachments/assets/4eb98889-8bff-4851-94e7-c4de084faad4" />I encountered another problem: when I changed Gradient Accumulation steps to 8, a training error occurred. How can I resolve this?

**Comment by songlin** [comment]
@kdygfxb  Please pull the latest code, it is fixed in the following code:

https://github.com/physical-superintelligence-lab/Psi0/blob/a5c304621071ba3de9b63d0f61e9ebd52a3e20ce/src/psi/trainers/finetune.py#L347-L349

**Comment by songlin** [comment]
Closing the issue for now, but feel free to reopen once have updates.
